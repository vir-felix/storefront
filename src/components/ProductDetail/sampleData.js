// Sample Catalog Product Data
export default {
  title: "Title",
  pageTitle: "SubTitle",
  description: "Description",
  vendor: "Vendor",
  price: {
    range: "$19.99"
  }
};
