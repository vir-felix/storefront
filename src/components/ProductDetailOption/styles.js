export default (theme) => ({
  variantOptionContainer: {
    display: "flex",
    width: `${theme.spacing.unit * 3.125}%`,
    maxWidth: `${theme.spacing.unit * 3.125}%`,
    padding: theme.spacing.unit * 1.25
  },
  optionButton: {
    "flex": "1 1 auto",
    "fontWeight": 600,
    "padding": theme.spacing.unit * 1.25,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.borderColor,
    "&:hover": {
      backgroundColor: theme.palette.reaction.activeElementBackground
    }
  },
  optionText: {
    fontWeight: 500,
    fontSize: "14px"
  },
  isSelected: {
    backgroundColor: theme.palette.reaction.activeElementBackground
  }
});
