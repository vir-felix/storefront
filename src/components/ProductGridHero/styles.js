export default (theme) => ({
  heroImg: {
    width: "100%",
    height: "325px",
    objectFit: "cover"
  },
  productGridContainer: {
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
