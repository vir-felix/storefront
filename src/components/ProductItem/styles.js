export const styles = (theme) => ({
  "@keyframes spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  "root": {},
  "productInfo": {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 10
  },
  "productMedia": {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  "img": {
    height: "auto",
    width: "100%"
  },
  "imgLoading": {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  "loadingIcon": {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  "link": {
    ...theme.typography.body2,
    "textAlign": "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});
