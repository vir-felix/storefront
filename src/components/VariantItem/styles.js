export default (theme) => ({
  variantButton: {
    "display": "flex",
    "justifyContent": "space-between",
    "textTransform": "none",
    "width": "100%",
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.contrastText,
    "paddingTop": theme.spacing.unit * 1.25,
    "paddingBottom": theme.spacing.unit * 1.25,
    "paddingLeft": theme.spacing.unit * 2.5,
    "paddingRight": theme.spacing.unit * 2.5,
    "&:hover": {
      border: "1px solid",
      borderColor: theme.palette.reaction.activeElementBorderColor,
      backgroundColor: theme.palette.primary.contrastText
    }
  },
  activeVariant: {
    border: "1px solid",
    borderColor: theme.palette.reaction.activeElementBorderColor
  }
});
