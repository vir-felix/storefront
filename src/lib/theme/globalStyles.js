export default
<style global jsx>
  {`
    a {
      color: inherit;
      text-decoration: none;
    }
  `}
</style>;
