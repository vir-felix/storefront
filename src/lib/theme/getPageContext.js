import { SheetsRegistry } from "jss";
import { createGenerateClassName } from "@material-ui/core/styles";
import theme from "./reactionTheme";

/**
 * @summary Creates and returns a page context object
 * @returns {Object} Page context object
 */
function createPageContext() {
  return {
    theme,
    // This is needed in order to deduplicate the injection of CSS in the page.
    sheetsManager: new Map(),
    // This is needed in order to inject the critical CSS.
    sheetsRegistry: new SheetsRegistry(),
    // The standard class name generator.
    generateClassName: createGenerateClassName()
  };
}

/**
 * Either creates a new or returns an existing page context object. If called from Node, creates
 * a new object every time. Otherwise creates and caches an object in the browser and
 * subsequent calls will return that.
 *
 * @summary Gets a page context object
 * @returns {Object} Page context object
 */
export default function getPageContext() {
  // Make sure to create a new context for every server-side request so that data
  // isn't shared between connections (which would be bad).
  if (!process.browser) {
    return createPageContext();
  }

  // Reuse context on the client-side.
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
