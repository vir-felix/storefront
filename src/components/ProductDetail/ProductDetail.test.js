import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "material-ui/styles";

import theme from "lib/theme/reactionTheme";
import ProductDetail from "./ProductDetail";
import sampleData from "./__mocks__/productData.mock";

test("basic snapshot", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <ProductDetail catalogProduct={sampleData} />
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
