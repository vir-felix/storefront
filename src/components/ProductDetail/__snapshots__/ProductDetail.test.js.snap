// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="inject-WithTracking-ProductDetail--with-routingStore-uiStore-root-1"
>
  <div
    className="MuiGrid-container-5 MuiGrid-spacing-xs-24-28 inject-WithTracking-ProductDetail--with-routingStore-uiStore-pdpContainer-2"
  >
    <div
      className="MuiGrid-item-6 MuiGrid-grid-xs-12-43 inject-WithTracking-ProductDetail--with-routingStore-uiStore-breadcrumbGrid-4"
    >
      <div
        className="Breadcrumbs-container-96"
      >
        <a
          className="WithTracking-Link--link-99"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="link"
          tabIndex={0}
          tracking={
            Object {
              "getTrackingData": [Function],
              "trackEvent": [Function],
            }
          }
        >
          <span
            className="Breadcrumbs-breadcrumbLink-97"
          >
            Home
          </span>
        </a>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-100 Breadcrumbs-breadcrumbIcon-98"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
          />
        </svg>
        <a
          className="WithTracking-Link--link-99"
          href="/tag/tag-a"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="link"
          tabIndex={0}
          tracking={
            Object {
              "getTrackingData": [Function],
              "trackEvent": [Function],
            }
          }
        >
          <span
            className="Breadcrumbs-breadcrumbLink-97"
          >
            Tag A
          </span>
        </a>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-100 Breadcrumbs-breadcrumbIcon-98"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
          />
        </svg>
        <a
          className="WithTracking-Link--link-99"
          href="/product/example-product"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="link"
          tabIndex={0}
          tracking={
            Object {
              "getTrackingData": [Function],
              "trackEvent": [Function],
            }
          }
        >
          <span
            className="Breadcrumbs-breadcrumbLink-97"
          >
            Reaction Sample Product
          </span>
        </a>
      </div>
    </div>
    <div
      className="MuiGrid-item-6 MuiGrid-grid-xs-12-43 MuiGrid-grid-sm-6-50"
    >
      <div
        className="inject-WithTracking-ProductDetail--with-routingStore-uiStore-section-3"
      >
        <div
          className="MuiGrid-container-5 MediaGallery-root-106"
        >
          <div
            className="MuiGrid-item-6 MuiGrid-grid-xs-12-43 MuiGrid-grid-sm-12-56"
          >
            <div
              className="MediaGallery-featured-107"
            >
              <div
                className="Img-imgWrapper-109 "
              >
                <img
                  alt=""
                  className="Img-img-111 Img-imgLoading-113 "
                  src="/resources/placeholder.gif"
                />
              </div>
            </div>
            <div
              className="MuiGrid-container-5 MuiGrid-spacing-xs-8-26"
            >
              <div
                className="MuiGrid-item-6 MuiGrid-grid-xs-3-34 MuiGrid-grid-sm-2-46"
              >
                <button
                  className="MuiButtonBase-root-116 MediaGalleryItem-root-115"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <div
                    className="Img-imgWrapper-109 "
                  >
                    <img
                      alt=""
                      className="Img-img-111 Img-imgLoading-113 "
                      src="/resources/placeholder.gif"
                    />
                  </div>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
              <div
                className="MuiGrid-item-6 MuiGrid-grid-xs-3-34 MuiGrid-grid-sm-2-46"
              >
                <button
                  className="MuiButtonBase-root-116 MediaGalleryItem-root-115"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <div
                    className="Img-imgWrapper-109 "
                  >
                    <img
                      alt=""
                      className="Img-img-111 Img-imgLoading-113 "
                      src="/resources/placeholder.gif"
                    />
                  </div>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
              <div
                className="MuiGrid-item-6 MuiGrid-grid-xs-3-34 MuiGrid-grid-sm-2-46"
              >
                <button
                  className="MuiButtonBase-root-116 MediaGalleryItem-root-115"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <div
                    className="Img-imgWrapper-109 "
                  >
                    <img
                      alt=""
                      className="Img-img-111 Img-imgLoading-113 "
                      src="/resources/placeholder.gif"
                    />
                  </div>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
              <div
                className="MuiGrid-item-6 MuiGrid-grid-xs-3-34 MuiGrid-grid-sm-2-46"
              >
                <button
                  className="MuiButtonBase-root-116 MediaGalleryItem-root-115"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <div
                    className="Img-imgWrapper-109 "
                  >
                    <img
                      alt=""
                      className="Img-img-111 Img-imgLoading-113 "
                      src="/resources/placeholder.gif"
                    />
                  </div>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
              <div
                className="MuiGrid-item-6 MuiGrid-grid-xs-3-34 MuiGrid-grid-sm-2-46"
              >
                <button
                  className="MuiButtonBase-root-116 MediaGalleryItem-root-115"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <div
                    className="Img-imgWrapper-109 "
                  >
                    <img
                      alt=""
                      className="Img-img-111 Img-imgLoading-113 "
                      src="/resources/placeholder.gif"
                    />
                  </div>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
              <div
                className="MuiGrid-item-6 MuiGrid-grid-xs-3-34 MuiGrid-grid-sm-2-46"
              >
                <button
                  className="MuiButtonBase-root-116 MediaGalleryItem-root-115"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <div
                    className="Img-imgWrapper-109 "
                  >
                    <img
                      alt=""
                      className="Img-img-111 Img-imgLoading-113 "
                      src="/resources/placeholder.gif"
                    />
                  </div>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
              <div
                className="MuiGrid-item-6 MuiGrid-grid-xs-3-34 MuiGrid-grid-sm-2-46"
              >
                <button
                  className="MuiButtonBase-root-116 MediaGalleryItem-root-115"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <div
                    className="Img-imgWrapper-109 "
                  >
                    <img
                      alt=""
                      className="Img-img-111 Img-imgLoading-113 "
                      src="/resources/placeholder.gif"
                    />
                  </div>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
              <div
                className="MuiGrid-item-6 MuiGrid-grid-xs-3-34 MuiGrid-grid-sm-2-46"
              >
                <button
                  className="MuiButtonBase-root-116 MediaGalleryItem-root-115"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <div
                    className="Img-imgWrapper-109 "
                  >
                    <img
                      alt=""
                      className="Img-img-111 Img-imgLoading-113 "
                      src="/resources/placeholder.gif"
                    />
                  </div>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="inject-WithTracking-ProductDetail--with-routingStore-uiStore-section-3"
      >
        <section>
          <h2
            className="MuiTypography-root-128 MuiTypography-title-134 TagGrid-title-126"
          >
            Tags
          </h2>
          <div
            className="MuiGrid-container-5 MuiGrid-spacing-xs-8-26"
          >
            <div
              className="MuiGrid-item-6"
            >
              <a
                className="WithTracking-Link--link-99"
                href="/tag/shoes"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="link"
                tabIndex={0}
                tracking={
                  Object {
                    "getTrackingData": [Function],
                    "trackEvent": [Function],
                  }
                }
              >
                <div
                  className="MuiChip-root-152 TagGrid-chip-127"
                  onClick={undefined}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={-1}
                >
                  <span
                    className="MuiChip-label-157"
                  >
                    Shoes
                  </span>
                </div>
              </a>
            </div>
            <div
              className="MuiGrid-item-6"
            >
              <a
                className="WithTracking-Link--link-99"
                href="/tag/shop"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="link"
                tabIndex={0}
                tracking={
                  Object {
                    "getTrackingData": [Function],
                    "trackEvent": [Function],
                  }
                }
              >
                <div
                  className="MuiChip-root-152 TagGrid-chip-127"
                  onClick={undefined}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={-1}
                >
                  <span
                    className="MuiChip-label-157"
                  >
                    Shop
                  </span>
                </div>
              </a>
            </div>
          </div>
        </section>
      </div>
    </div>
    <div
      className="MuiGrid-item-6 MuiGrid-grid-xs-12-43 MuiGrid-grid-sm-6-50"
    >
      <div
        className="MuiGrid-item-6 MuiGrid-grid-sm-12-56"
      >
        <h1
          className="MuiTypography-root-128 MuiTypography-display2-131 MuiTypography-gutterBottom-145"
        >
          Reaction Sample Product
        </h1>
      </div>
      <div
        className="MuiGrid-item-6 MuiGrid-grid-sm-12-56"
      >
        <div
          className="MuiTypography-root-128 MuiTypography-title-134 MuiTypography-gutterBottom-145"
        >
          $12.99 - $19.99
        </div>
        <p
          className="MuiTypography-root-128 MuiTypography-body1-137"
        >
          Example Manufacturer
        </p>
        <p
          className="MuiTypography-root-128 MuiTypography-body1-137"
        >
          Sign in as administrator to edit.
You can clone this product from the product grid.
You can upload images click or drag in image box on the left here.
Tag this product below, and then add tag in navigation.
Click the bookmark in the tag to set product url.
Option variants, price, quantity, and child variants are created by clicking on the variant below, clone the variant to add more options.
Details can be added below the image for more specific product information.
 Login next to the cart, and then click the dashboard icon for more tools.
        </p>
      </div>
      <div
        className="VariantList-variantsContainer-159"
      >
        <div
          className="VariantList-variantItem-160"
        >
          <button
            className="MuiButtonBase-root-116 VariantItem-variantButton-162 VariantItem-activeVariant-163"
            disabled={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiTypography-root-128 MuiTypography-body1-137"
            >
              Sample Variant
            </span>
            <span
              className="MuiTypography-root-128 MuiTypography-body1-137"
            >
              $12.99 - $19.99
            </span>
          </button>
          <div
            className="VariantList-alert-161"
          >
            <div
              className="BadgeOverlay-badgeOverlay-164 BadgeOverlay-faded-165"
            />
          </div>
        </div>
        <div
          className="VariantList-variantItem-160"
        >
          <button
            className="MuiButtonBase-root-116 VariantItem-variantButton-162"
            disabled={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiTypography-root-128 MuiTypography-body1-137"
            >
              Sample Variant 2
            </span>
            <span
              className="MuiTypography-root-128 MuiTypography-body1-137"
            >
              $12.99 - $19.99
            </span>
          </button>
        </div>
        <div
          className="Divider-container-175"
        >
          <hr
            className="Divider-item-177"
          />
          <span
            className="MuiTypography-root-128 MuiTypography-body1-137 Divider-label-176"
          >
            Available Options
          </span>
          <hr
            className="Divider-item-177"
          />
        </div>
        <div
          className="MuiGrid-container-5 MuiGrid-spacing-xs-8-26 OptionsList-root-178"
        >
          <div
            className="MuiGrid-item-6"
          >
            <button
              className="MuiButtonBase-root-116 ProductDetailOption-optionButton-181"
              disabled={undefined}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiTypography-root-128 MuiTypography-body1-137 ProductDetailOption-optionText-182"
              >
                Red
              </span>
            </button>
            <div
              className="OptionsList-alert-179"
            >
              <div
                className="BadgeOverlay-badgeOverlay-164"
              />
            </div>
          </div>
          <div
            className="MuiGrid-item-6"
          >
            <button
              className="MuiButtonBase-root-116 ProductDetailOption-optionButton-181"
              disabled={undefined}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiTypography-root-128 MuiTypography-body1-137 ProductDetailOption-optionText-182"
              >
                Green
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-container-5"
      >
        <div
          className="MuiGrid-item-6 MuiGrid-grid-xs-12-43 inject-ProductDetailAddToCart-with-uiStore-quantityGrid-188"
        >
          <div
            className="Divider-container-175"
          >
            <hr
              className="Divider-item-177"
            />
            <hr
              className="Divider-item-177"
            />
          </div>
          <span
            className="MuiTypography-root-128 MuiTypography-body1-137 inject-ProductDetailAddToCart-with-uiStore-quantityTypography-190"
          >
            Quantity
          </span>
          <div
            aria-describedby={undefined}
            className="MuiFormControl-root-191"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              className="MuiInput-root-195 inject-ProductDetailAddToCart-with-uiStore-quantityContainer-187 MuiInput-formControl-196"
            >
              <div
                className="MuiInputAdornment-root-208"
              >
                <button
                  className="MuiButtonBase-root-116 inject-ProductDetailAddToCart-with-uiStore-incrementButton-186"
                  color="default"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                  variant="outlined"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-100"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19,13H5V11H19V13Z"
                    />
                  </svg>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
              <input
                aria-invalid={false}
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-203 inject-ProductDetailAddToCart-with-uiStore-quantityInput-189"
                defaultValue={undefined}
                disabled={false}
                id="addToCartQuantityInput"
                name={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="text"
                value={1}
              />
              <div
                className="MuiInputAdornment-root-208"
              >
                <button
                  className="MuiButtonBase-root-116 inject-ProductDetailAddToCart-with-uiStore-incrementButton-186"
                  color="default"
                  disabled={undefined}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                  variant="outlined"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-100"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                    />
                  </svg>
                  <span
                    className="MuiTouchRipple-root-119"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-6 MuiGrid-grid-xs-12-43"
        >
          <button
            className="MuiButtonBase-root-116 inject-ProductDetailAddToCart-with-uiStore-addToCartButton-184"
            disabled={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiTypography-root-128 MuiTypography-body1-137 inject-ProductDetailAddToCart-with-uiStore-addToCartText-185"
            >
              Add to cart
            </span>
            <span
              className="MuiTouchRipple-root-119"
            />
          </button>
        </div>
      </div>
      <div
        className="HiddenCss-mdUp-226"
      >
        <div
          className="inject-CartPopover-with-uiStore-container-211 inject-CartPopover-with-uiStore-isContainerHidden-213"
        >
          <div
            className="MuiGrid-container-5 MuiGrid-spacing-xs-24-28 inject-CartPopover-with-uiStore-gridContainer-212"
          >
            <div
              className="MuiGrid-item-6 MuiGrid-grid-xs-12-43 inject-CartPopover-with-uiStore-containerItem-215"
            >
              <img
                alt="Item Title"
                className="inject-CartPopover-with-uiStore-addedToCartImg-216"
                src="http://localhost:3000/assets/files/Media/nbYKLrZST5DSF87md/thumbnail/chuttersnap-265339-unsplash.png"
              />
              <span
                className="MuiTypography-root-128 MuiTypography-body1-137 inject-CartPopover-with-uiStore-addedToCartText-218"
              >
                10
                 "
                <span
                  className="inject-CartPopover-with-uiStore-addedToCartItemName-217"
                >
                  Item Title
                </span>
                " added to cart
              </span>
            </div>
            <div
              className="MuiGrid-item-6 MuiGrid-grid-xs-12-43"
            >
              <a
                className="WithTracking-Link--link-99"
                href="/checkout"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="link"
                tabIndex={0}
                tracking={
                  Object {
                    "getTrackingData": [Function],
                    "trackEvent": [Function],
                  }
                }
              >
                <div
                  className="Button__ButtonDiv-bnfrK eloVkm"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex={0}
                  title={undefined}
                >
                  <div>
                    Checkout
                  </div>
                  <div
                    className="Button__SpinnerWrap-bgOcuM cCKgPo"
                    style={
                      Object {
                        "opacity": 0,
                        "paddingLeft": 0,
                        "width": 0,
                      }
                    }
                  >
                    <svg
                      className="spinner__SpinnerSvg-iBTUvT bunzNs"
                      height="40px"
                      style={
                        Object {
                          "enableBackground": "new 0 0 50 50",
                        }
                      }
                      version="1.1"
                      viewBox="0 0 50 50"
                      width="40px"
                      x="0px"
                      y="0px"
                    >
                      <path
                        d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
                        fill="#000"
                      >
                        <animateTransform
                          attributeName="transform"
                          attributeType="xml"
                          dur="1s"
                          from="0 25 25"
                          repeatCount="indefinite"
                          to="360 25 25"
                          type="rotate"
                        />
                      </path>
                    </svg>
                  </div>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
